src/zql/ivm/graph/operators/concat-operator.test.ts

```diff
@@ -0,0 +1,73 @@
+import {expect, test} from 'vitest';
+import {Multiset} from '../../multiset.js';
+import {DifferenceStream} from '../difference-stream.js';
+
+test('All branches notify', () => {
+  type T = {x: number};
+  const inputs = [
+    new DifferenceStream<T>(),
+    new DifferenceStream<T>(),
+    new DifferenceStream<T>(),
+  ];
+  const output = inputs[0].concat(...inputs.slice(1));
+
+  let version = 1;
+
+  const items: Multiset<T>[] = [];
+  output.debug((v, d) => {
+    expect(v).toBe(version);
+    items.push(d);
+  });
+
+  inputs[0].newData(version, [
+    [{x: 1}, 1],
+    [{x: 2}, 2],
+  ]);
+  inputs[0].commit(version);
+
+  expect(items).toEqual([
+    [
+      [{x: 1}, 1],
+      [{x: 2}, 2],
+    ],
+  ]);
+
+  items.length = 0;
+  version++;
+
+  inputs[0].newData(version, [[{x: 0}, 1]]);
+  inputs[1].newData(version, [[{x: 1}, 1]]);
+  inputs[2].newData(version, [[{x: 2}, 2]]);
+  inputs[0].commit(version);
+  inputs[1].commit(version);
+  inputs[2].commit(version);
+  expect(items).toEqual([[[{x: 0}, 1]], [[{x: 1}, 1]], [[{x: 2}, 2]]]);
+});
+
+test('Test with single input', () => {
+  type T = {x: number};
+  const input = new DifferenceStream<T>();
+
+  const output = input.concat();
```

Comment:
does it make sense to concat with nothing?

Reply:
Not really. We can enforce the length using types.

I might remove this method and provide a function instead. As you can see, I split the array and then join it later which useless work with the sole purpose of fitting this API.

Reply:
Removed method in favor of function.

I didn't add any code/or types to ensure the length. `concat([])` and `concat([s])` works as expected even if not very useful.